test/host/wasi_nn/wasi_nn.cpp

```diff
@@ -0,0 +1,323 @@
+        WasiNNSetInput.run(&MemInst,
+                           std::initializer_list<WasmEdge::ValVariant>{
+                               UINT32_C(0), UINT32_C(10), SetInputEntryPtr},
+                           Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(),
+              static_cast<uint32_t>(ErrNo::InvalidArgument));
+  }
+  // tensor type not FP32
+  BuilderPtr = SetInputEntryPtr;
+  writeFatPointer(MemInst, StorePtr, TensorDim.size(), BuilderPtr);
+  writeUInt32(MemInst, UINT32_C(2), BuilderPtr);
+  writeFatPointer(MemInst, StorePtr + TensorDim.size() * 4, TensorData.size(),
+                  BuilderPtr);
+  {
+    EXPECT_TRUE(
+        WasiNNSetInput.run(&MemInst,
+                           std::initializer_list<WasmEdge::ValVariant>{
+                               UINT32_C(0), UINT32_C(0), SetInputEntryPtr},
+                           Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(),
+              static_cast<uint32_t>(ErrNo::InvalidArgument));
+  }
+  // successful run
+  BuilderPtr = SetInputEntryPtr;
+  writeFatPointer(MemInst, StorePtr, TensorDim.size(), BuilderPtr);
+  writeUInt32(MemInst, UINT32_C(1), BuilderPtr);
+  writeFatPointer(MemInst, StorePtr + TensorDim.size() * 4, TensorData.size(),
+                  BuilderPtr);
+  {
+    EXPECT_TRUE(
+        WasiNNSetInput.run(&MemInst,
+                           std::initializer_list<WasmEdge::ValVariant>{
+                               UINT32_C(1), UINT32_C(0), SetInputEntryPtr},
+                           Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Success));
+  }
+  StorePtr += (TensorDim.size() * 4 + TensorData.size());
+
+  // compute test
+  WasmEdge::Host::WasiNNCompute WasiNNCompute(Env);
+  WasmEdge::Host::WasiNNCompute EmptyWasiNNCompute(EmptyEnv);
+  // context id exceed
+  {
+    EXPECT_TRUE(WasiNNCompute.run(
+        &MemInst, std::initializer_list<WasmEdge::ValVariant>{UINT32_C(3)},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(),
+              static_cast<uint32_t>(ErrNo::InvalidArgument));
+  }
+  // empty ctx compute
+  {
+    EXPECT_TRUE(EmptyWasiNNCompute.run(
+        &MemInst, std::initializer_list<WasmEdge::ValVariant>{UINT32_C(0)},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Busy));
+  }
+  // successful run
+  {
+    EXPECT_TRUE(WasiNNCompute.run(
+        &MemInst, std::initializer_list<WasmEdge::ValVariant>{UINT32_C(1)},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Success));
+  }
+
+  WasmEdge::Host::WasiNNGetOuput WasiNNGetOuput(Env);
+  // WasmEdge::Host::WasiNNGetOuput EmptyWasiNNGetOuput(EmptyEnv);
+  // no compute ctx request
+  {
+    EXPECT_TRUE(WasiNNGetOuput.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            UINT32_C(0), UINT32_C(0), StorePtr, 65532, BuilderPtr},
+        Errno));
+    // FIXME should allow a request for output with no computation before?
```

Comment:
You can add this test for observing the results.

Reply:
I have added this test before. It seems like OpenVINO just returns a random results with no errors alert.